benches/nexmark/generator/mod.rs

```diff
@@ -58,6 +56,18 @@ impl<R: Rng> NexmarkGenerator<R> {
         self.events_count_so_far = 0;
     }
 
+    // Returns the real current wallclock time or a specific range of numbers
+    // in tests.
+    pub fn wallclock_time(&mut self) -> u64 {
+        match &mut self.wallclock_iterator {
+            Some(i) => i.next().unwrap(),
+            None => SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .unwrap()
+                .as_millis() as u64,
+        }
```

Comment:
```suggestion
        self.wallclock_iterator
            .as_mut()
            .map(|wall| wall.next().unwrap())
            .unwrap_or_else(|| {
                SystemTime::now()
                    .duration_since(SystemTime::UNIX_EPOCH)
                    .unwrap()
                    .as_millis() as u64
            })
```

Reply:
My brain seems to parse the simple match more easily here (no mapping or extra unwrapping), so I'm interested to know if that's just my less experience or whether there's a benefit to not using a match even in this simple case?

Reply:
I'm with @absoludity on this one. Sometimes a match can be replaced with a single operator, eg map or unwrap_or_else. But in this case a chain of two transformations is harder to parse imo.